GMX-Bots/Large-Open-and-Increase-Positions/src/agent.ts

```diff
@@ -0,0 +1,59 @@
+import { Finding, getEthersProvider, HandleTransaction, LogDescription, TransactionEvent } from "forta-agent";
+import { providers } from "ethers";
+import BigNumber from "bignumber.js";
+import { createFinding } from "./utils";
+import NetworkData, { NETWORK_MAP } from "./network";
+import { INCREASE_POSITION_EVENT, UPDATE_POSITION_EVENT, PRICE_MULTIPLIER } from "./constants";
+import NetworkManager from "./network";
+
+const networkManager = new NetworkManager(NETWORK_MAP);
+
+export const initialize = (provider: providers.Provider) => {
+  return async () => {
+    const { chainId } = await provider.getNetwork();
+    networkManager.setNetwork(chainId);
+  };
+};
+
+export const provideTransactionHandler = (
+  networkManager: NetworkData,
+  updatePositionEvent: string,
+  increasePositionEvent: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    // filter transaction logs for increase and update position events
+    const eventLogs: LogDescription[] = txEvent.filterLog(
+      [increasePositionEvent, updatePositionEvent],
+      networkManager.vaultAddress
+    );
+
+    // extract only increaseEventLogs
+    const increaseEventLogs: LogDescription[] = eventLogs.filter((eventLogs) => eventLogs.name == "IncreasePosition");
+
+    // extract only updateEventLogs
+    const updateEventLogs: LogDescription[] = eventLogs.filter((eventLogs) => eventLogs.name == "UpdatePosition");
+
+    increaseEventLogs.forEach((increaseEventLog, index) => {
+      const updateEventLog = updateEventLogs[index];
+      const { sizeDelta, key, account } = increaseEventLog.args;
+      const { size } = updateEventLog.args;
+      const baseSizeDelta: BigNumber = new BigNumber(sizeDelta.toString()).dividedBy(
+        new BigNumber(PRICE_MULTIPLIER.toString())
+      );
```

Comment:
Can you explain why the `sizeDelta` needs to be divided bt `PRICE_MULTIPLIER`? Readme says that the threshold is for the size, but you have this division here, so I want to understand why to be able to evaluate if the threshold value is clear or not.

Reply:
In the [docs](https://gmxio.gitbook.io/gmx/contracts#opening-increasing-a-position),  it is specified that `(10 ** 30)` should be used to accurately determine the sizeDelta and since I intend to compare this  value with the `threshold`, I used Bignumber's `gt` method to check if this derived sizeDelta is greater than the threshold